tachyon/math/base/semigroups.h

```diff
@@ -141,6 +141,32 @@ class MultiplicativeSemigroup {
   template <typename Scalar>
   [[nodiscard]] constexpr auto Pow(const Scalar& scalar) const {
     if constexpr (std::is_constructible_v<BigInt<1>, Scalar>) {
+      const G& g = static_cast<const G&>(*this);
+      if (scalar == 0) {
+        return MulResult::One();
+      } else if (scalar == 1) {
+        return g;
+      } else if (scalar == 2) {
+        return Square();
+      } else if (scalar == 3) {
+        return Square() * g;
+      } else if (scalar == 4) {
+        return Square().Square();
+      } else if (scalar == 5) {
+        MulResult g4 = Square();
+        g4.SquareInPlace();
+        return g4 * g;
+      } else if (scalar == 6) {
+        MulResult g2 = Square();
+        MulResult g4 = g2;
+        g2.SquareInPlace();
+        return g4 * g2;
+      } else if (scalar == 7) {
+        MulResult g2 = Square();
+        MulResult g4 = g2;
+        g4.SquareInPlace();
+        return g4 * g2 * g;
+      }
```

Comment:
It seems like DoPow already works this way. Has this implementation been effective?

Reply:
Yes, since it doesn't compute bit vector of the exponent.